src/journal/managed-journal-file.c

```diff
@@ -538,6 +539,14 @@ int managed_journal_file_open_reliably(
         /* The file is corrupted. Rotate it away and try it again (but only once) */
         log_warning_errno(r, "File %s corrupted or uncleanly shut down, renaming and replacing.", fname);
 
+        if (!template) {
+                /* The file is corrupted and no template is specified. Try opening it read-only as the
+                 * template before rotating to inherit its sequence number and ID. */
+                managed_journal_file_open(-1, fname, O_RDONLY, file_flags, 0, compress_threshold_bytes, NULL,
+                                   mmap_cache, deferred_closes, NULL, &old_file);
```

Comment:
Missing error handling, and spurious indentation.
(Please also fix the indentation below.)